detection_rules/rule.py

```diff
@@ -675,10 +684,35 @@ def _post_dict_transform(self, obj: dict) -> dict:
         subclass.from_dict(obj)
         return obj
 
-    def add_related_integrations(self, obj: dict) -> None:
+    def _add_related_integrations(self, obj: dict) -> None:
         """Add restricted field related_integrations to the obj."""
-        # field_name = "related_integrations"
-        ...
+        field_name = "related_integrations"
+        package_integrations = obj.get(field_name, [])
+
+        if not package_integrations and self.metadata.integration:
```

Comment:
I updated this and L715 to _only_ set the field if a value is set (explicitly or dynamically) to avoid inserting empty arrays